lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -32,34 +32,37 @@ defmodule Mix.Compilers.Elixir do
     # change to files are still picked up by the compiler. This
     # timestamp is used when writing BEAM files and the manifest.
     timestamp = :calendar.universal_time()
-    all = Mix.Utils.extract_files(srcs, [:ex])
+    all_paths = MapSet.new(Mix.Utils.extract_files(srcs, [:ex]))
 
     {all_modules, all_sources} = parse_manifest(manifest, dest)
     modified = Mix.Utils.last_modified(manifest)
+    prev_sources_paths =
+      for source(source: source) <- all_sources, into: MapSet.new(), do: source
 
     removed =
-      for source(source: source) <- all_sources,
-          not(source in all),
-          do: source
+      prev_sources_paths
```

Comment:
I think we can drop `_sources_` part from the name (will be consistent with `all_paths` and `new_paths`). :bowtie: